assets/js/googlesitekit/notifications/util/shouldNotificationBeAddedToQueue.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Site Kit by Google, Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { isFeatureEnabled } from '../../../features';
+
+/**
+ * Checks if a notification should be added to the queue based on
+ * its feature flags, groupID, viewContext, etc.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object}   notification            Notification object.
+ * @param {string}   groupID                 Group ID to check against the notification's groupID.
+ * @param {string}   viewContext             View context to check against the notification's viewContexts (if present).
+ * @param {Function} isNotificationDismissed Function to check if a notification is dismissed.
+ * @return {boolean} Returns true if the notification should be added to the queue, false otherwise.
+ */
+export function shouldNotificationBeAddedToQueue(
+	notification,
+	groupID,
+	viewContext,
+	isNotificationDismissed = () => false
```

Comment:
WDYT about grouping these into an object? It would help making cases like this a bit more clear
 
<img width="461" alt="image" src="https://github.com/user-attachments/assets/c28d3277-957c-478b-ae0c-3f524c3f642a" />

Reply:
I thought about it too, so yes 👍🏻